provider/cmd/pulumi-resource-aws/main.go

```diff
@@ -19,16 +19,27 @@ package main
 import (
 	"context"
 	_ "embed"
+	"os"
 
 	aws "github.com/pulumi/pulumi-aws/provider/v6"
 	pf "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
 )
 
 //go:embed schema-embed.json
 var pulumiSchema []byte
 
+//go:embed schema-light-embed.json
+var pulumiSchemaLight []byte
```

Comment:
https://github.com/pulumi/pulumi-aws/issues/4383 roughly 750MB for go libs and 50MB for schema, this would add another 50mb on top. It's not pleasant but I think we can absorb this.

Reply:
We could gzip it possibly and decompress on-demand.

Reply:
50 over 750 is a material tax that every user will pay for a feature for a few. My gut reaction is to vote against this and service the minimal version only if it's proven to be safe based on non-node SDK generation.

Reply:
Let me look at a few things. Dropping descriptions and gzipping can be easy to try and can reduce the tax.

Reply:
>  it's proven to be safe based on non-node SDK generation.

This seems to hold. It's safe.

Reply:
```
du -sh schema-minimal-embed.json.gz                                                                                                                                                                                                      ~/code/pulumi-aws/provider/cmd/pulumi-resource-aws
2.8M    schema-minimal-embed.json.gz 
```

Reply:
Even with descriptions it compresses well. Decompression will only tax based on opt-in. WDYT?

Reply:
That sounds good to me